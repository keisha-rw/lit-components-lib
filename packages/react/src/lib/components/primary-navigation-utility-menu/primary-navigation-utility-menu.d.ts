import * as React from 'react';
export declare const PdsPrimaryNavigationUtilityMenu: React.FunctionComponent<Partial<{
    variant: "inverted" | "default";
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
