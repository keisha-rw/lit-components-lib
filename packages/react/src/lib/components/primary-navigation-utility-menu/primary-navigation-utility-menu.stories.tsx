import * as React from 'react';
import type { StoryFn } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { PdsPrimaryNavigationUtilityMenu } from './primary-navigation-utility-menu';
import { PdsPrimaryNavigationUtilityMenuItem } from '../primary-navigation-utility-menu-item/primary-navigation-utility-menu-item';

export default {
  title: 'Components/Primary navigation/Primary navigation utility menu',
  component: PdsPrimaryNavigationUtilityMenu,
  tags: ['autodocs'],
  parameters: {
    happo: false,
    docs: {
      description: {
        component:
          'Note: This component is ONLY meant to be used as a subcomponent of the primary navigation component.',
      },
    },
  },
};

const Template: StoryFn<typeof PdsPrimaryNavigationUtilityMenu> = () => (
  <PdsPrimaryNavigationUtilityMenu slot="primary-navigation-utility-menu">
    <PdsPrimaryNavigationUtilityMenuItem
      href="#"
      onHandleClick={(e: any) =>
        action('pds-primary-navigation-utility-menu-item-click')(e.detail)
      }
    >
      Link
    </PdsPrimaryNavigationUtilityMenuItem>
  </PdsPrimaryNavigationUtilityMenu>
);
export const Default = Template.bind({});
Default.args = {};
