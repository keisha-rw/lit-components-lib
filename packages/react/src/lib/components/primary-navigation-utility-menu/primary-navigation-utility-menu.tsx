import * as React from 'react';
import { PdsPrimaryNavigationUtilityMenu as PdsPrimaryNavigationUtilityMenuElement } from '@keisha/design-system';
import { PdsPrimaryNavigationUtilityMenuProps } from '../../built-component-props';
import { createPdsReactComponent } from '../../create-pds-react-component';

export const PdsPrimaryNavigationUtilityMenu =
  createPdsReactComponent<PdsPrimaryNavigationUtilityMenuProps>()({
    tagName: 'pds-primary-navigation-utility-menu',
    elementClass: PdsPrimaryNavigationUtilityMenuElement,
    react: React,
  });
