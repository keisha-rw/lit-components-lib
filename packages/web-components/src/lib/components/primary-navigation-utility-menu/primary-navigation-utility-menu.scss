.pds-c-primary-navigation-utility-menu__divider {
  padding: 0 size(3) size(5) size(3);

  @media all and (min-width: $pds-bp-lg) {
    display: none;
  }

  @media all and (width < $pds-bp-lg) {
    padding: size(3);
  }
}

.pds-c-primary-navigation-utility-menu {
  float: none;
  display: block;
  padding: 0 size(3);

  @media all and (min-width: $pds-bp-lg) {
    float: right;
    display: inline;
    border: none;
    padding: size(1.5);
  }
}

.pds-c-primary-navigation-utility-menu--default {
  --pds-primary-navigation-utility-menu-item-link-background: var(
    --pds-component-primary-nav-default-mobile-submenu-background
  );
  --pds-primary-navigation-utility-menu-item-link-color: var(
    --pds-component-primary-nav-default-mobile-submenu-label
  );
  --pds-primary-navigation-utility-menu-item-href-color: var(
    --pds-component-primary-nav-default-mobile-link
  );
  --pds-primary-navigation-utility-menu-item-href-hover-color: var(
    --pds-component-primary-nav-default-mobile-hover-link
  );
  --pds-primary-navigation-utility-menu-item-link-hover-background: var(
    --pds-component-primary-nav-default-mobile-submenu-hover-background
  );
  --pds-primary-navigation-utility-menu-item-link-hover-color: var(
    --pds-component-primary-nav-default-mobile-submenu-hover-label
  );
  --pds-primary-navigation-utility-menu-item-link-active-background: var(
    --pds-component-primary-nav-default-mobile-submenu-active-background
  );
  --pds-primary-navigation-utility-menu-item-link-active-color: var(
    --pds-component-primary-nav-default-mobile-submenu-active-label
  );

  @media all and (min-width: $pds-bp-lg) {
    --pds-primary-navigation-utility-menu-item-link-background: var(
      --pds-component-primary-nav-default-nav-item-background
    );
    --pds-primary-navigation-utility-menu-item-link-color: var(
      --pds-component-primary-nav-default-nav-item-text
    );
    --pds-primary-navigation-utility-menu-item-link-hover-background: var(
      --pds-component-primary-nav-default-nav-item-hover-background
    );
    --pds-primary-navigation-utility-menu-item-link-hover-color: var(
      --pds-component-primary-nav-default-nav-item-hover-text
    );
    --pds-primary-navigation-utility-menu-item-link-active-background: var(
      --pds-component-primary-nav-default-nav-item-active-background
    );
    --pds-primary-navigation-utility-menu-item-link-active-color: var(
      --pds-component-primary-nav-default-nav-item-active-text
    );
  }
}

.pds-c-primary-navigation-utility-menu--inverted {
  --pds-primary-navigation-utility-menu-item-link-background: var(
    --pds-component-primary-nav-inverted-mobile-submenu-background
  );
  --pds-primary-navigation-utility-menu-item-link-color: var(
    --pds-component-primary-nav-inverted-mobile-submenu-label
  );
  --pds-primary-navigation-utility-menu-item-href-color: var(
    --pds-component-primary-nav-inverted-mobile-link
  );
  --pds-primary-navigation-utility-menu-item-href-hover-color: var(
    --pds-component-primary-nav-inverted-mobile-hover-link
  );
  --pds-primary-navigation-utility-menu-item-link-hover-background: var(
    --pds-component-primary-nav-inverted-mobile-submenu-hover-label
  );
  --pds-primary-navigation-utility-menu-item-link-hover-color: var(
    --pds-component-primary-nav-inverted-mobile-submenu-hover-label
  );
  --pds-primary-navigation-utility-menu-item-link-active-background: var(
    --pds-component-primary-nav-inverted-mobile-submenu-active-label
  );
  --pds-primary-navigation-utility-menu-item-link-active-color: var(
    --pds-component-primary-nav-inverted-mobile-submenu-active-label
  );

  @media all and (min-width: $pds-bp-lg) {
    --pds-primary-navigation-utility-menu-item-link-background: var(
      --pds-component-primary-nav-inverted-nav-item-background
    );
    --pds-primary-navigation-utility-menu-item-link-color: var(
      --pds-component-primary-nav-inverted-nav-item-text
    );
    --pds-primary-navigation-utility-menu-item-link-hover-background: var(
      --pds-component-primary-nav-inverted-nav-item-hover-background
    );
    --pds-primary-navigation-utility-menu-item-link-hover-color: var(
      --pds-component-primary-nav-inverted-nav-item-hover-text
    );
    --pds-primary-navigation-utility-menu-item-link-active-background: var(
      --pds-component-primary-nav-inverted-nav-item-active-background
    );
    --pds-primary-navigation-utility-menu-item-link-active-color: var(
      --pds-component-primary-nav-inverted-nav-item-active-text
    );
  }
}

.pds-c-primary-navigation-utility-menu__list {
  display: flex;
  flex-direction: column;
  gap: size(1);
  margin: 0;
  padding: 0;
  list-style: none;

  @media all and (min-width: $pds-bp-lg) {
    flex-direction: row;
  }

  @media all and (width < $pds-bp-lg) {
    gap: size(3);
  }
}
