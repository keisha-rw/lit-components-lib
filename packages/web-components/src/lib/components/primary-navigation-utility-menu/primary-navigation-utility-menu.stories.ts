import { html } from 'lit';
import { Meta, StoryObj } from '@storybook/web-components';
import './primary-navigation-utility-menu';
import '../primary-navigation-utility-menu-item/primary-navigation-utility-menu-item';

const meta: Meta = {
  title: 'Components/Primary navigation/Primary navigation utility menu',
  component: 'pds-primary-navigation-utility-menu',
  tags: ['autodocs'],
  parameters: {
    happo: false,
    docs: {
      description: {
        component:
          'Note: This component is ONLY meant to be used as a subcomponent of the primary navigation component.',
      },
    },
  },
};
export default meta;

// span element is needed to wrap plain text slots https://github.com/storybookjs/storybook/issues/15963

export const Default: StoryObj = {
  name: 'Default',
  args: {
    text: 'Default',
  },
  render: () =>
    html`<pds-primary-navigation-utility-menu
      ><pds-primary-navigation-utility-menu-item href="#"
        >Link</pds-primary-navigation-utility-menu-item
      ></pds-primary-navigation-utility-menu
    >`,
};
