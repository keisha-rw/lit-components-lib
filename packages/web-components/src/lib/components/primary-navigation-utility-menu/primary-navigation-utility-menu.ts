import { html } from 'lit';
import { property } from 'lit/decorators.js';
import { pdsCustomElement as customElement } from '../../decorators/pds-custom-element';
import { PdsElement } from '../PdsElement';
import styles from './primary-navigation-utility-menu.scss?inline';
import '../hr/hr';

/**
 * @summary This component is a navigational element (nav) that holds an Un unordered list element (ul)
 *
 * @slot default Contains one or more menu items within the pds-primary-navigation-utility-menu, restricted to pds-primary-navigation-utility-menu-item elements
 */
// TODO: should this be a layout component?  Need to switch that.
@customElement('pds-primary-navigation-utility-menu', {
  category: 'component',
  type: 'component',
  state: 'stable',
  styles,
})
export class PdsPrimaryNavigationUtilityMenu extends PdsElement {
  /**
   * Style variant
   * - **default** renders the default primary-navigation-utility-menu
   * - **inverted** renders the inverted primary-navigation-utility-menu
   */
  @property()
  variant: 'default' | 'inverted' = 'default';

  /**
   * @internal
   */
  get classNames() {
    return {
      [this.variant]: !!this.variant,
    };
  }

  render() {
    return html`<pds-hr
        class="pds-c-primary-navigation-utility-menu__divider"
      ></pds-hr>
      <nav class=${this.getClass()}>
        <ul class="pds-c-primary-navigation-utility-menu__list" role="list">
          <slot
            allowed-elements="pds-primary-navigation-utility-menu-item"
            @slotchange=${this.handleSlotValidation}
          ></slot>
        </ul>
      </nav>`;
  }
}
