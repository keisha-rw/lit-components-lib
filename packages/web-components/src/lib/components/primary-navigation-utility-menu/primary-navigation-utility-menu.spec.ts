import * as path from 'path';
import initStoryshots from '@storybook/addon-storyshots';
import { assert, fixture } from '@open-wc/testing';
import { PdsPrimaryNavigationUtilityMenu } from './primary-navigation-utility-menu';
import createSnapshots from '../../../../test/utils/snapshots';

initStoryshots({
  storyKindRegex: /Primary navigation utility menu/,
  framework: 'web-components',
  configPath: path.join(__dirname, '../../../../.storybook'),
  asyncJest: true,
  test: createSnapshots,
});

describe('PrimaryNavigationUtilityMenu unit tests', () => {
  let element: Element;

  beforeEach(async () => {
    element = await fixture(`<pds-primary-navigation-utility-menu>
      <pds-primary-navigation-utility-menu-item href="#">Link</pds-primary-navigation-utility-menu-item>
    <pds-primary-navigation-utility-menu/>`);
  });

  it('is an instance of PdsPrimaryNavigationUtilityMenu', () => {
    expect(element).toBeInstanceOf(PdsPrimaryNavigationUtilityMenu);
  });

  it('is accessible', async () => {
    await assert.isAccessible(element, {
      ignoredRules: ['landmark-unique'],
    });
  });

  it('populates the slot correctly with type pds-primary-navigation-utility-menu-item', async () => {
    const consoleError = jest
      .spyOn(console, 'error')
      .mockImplementation(() => {});
    element = await fixture(
      '<pds-primary-navigation-utility-menu variant="default"><pds-primary-navigation-utility-menu-item href="#>Link</pds-primary-navigation-utility-menu-item></pds-primary-navigation-utility-menu>',
    );
    expect(consoleError).not.toHaveBeenCalled();
    consoleError.mockRestore();
  });

  it('does not populate the slot correctly with any element not type of pds-primary-navigation-utility-menu-item', async () => {
    const consoleError = jest
      .spyOn(console, 'error')
      .mockImplementation(() => {});
    element = await fixture(
      '<pds-primary-navigation-utility-menu variant="default"><span>Link</span></pds-primary-navigation-utility-menu>',
    );
    expect(element).toMatchSnapshot();
    expect(consoleError).toHaveBeenCalledTimes(1);
    consoleError.mockRestore();
  });
});
