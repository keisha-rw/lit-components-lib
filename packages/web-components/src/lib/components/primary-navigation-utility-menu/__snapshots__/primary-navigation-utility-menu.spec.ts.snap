// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryNavigationUtilityMenu unit tests does not populate the slot correctly with any element not type of pds-primary-navigation-utility-menu-item 1`] = `
<pds-primary-navigation-utility-menu
  variant="default"
/>
`;
